src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -903,6 +918,20 @@ public AppPage verifyHtmlMainContent(String filePath) throws IOException {
         return verifyHtmlPart(MAIN_CONTENT, filePath);
     }
 
+    public AppPage verifyHtmlMainContentWithReloadRetry(final String filePath) throws IOException {
+        return RetryManager.runUntilNoException(new RetryableTaskReturnsThrows<AppPage, IOException>("HTML verification") {
+            @Override
+            public AppPage run() throws IOException {
```

Comment:
Especially for multiple exception types that the user wants to catch, the only way to pass would be through the (normal non-generic) parameter. A method `runUntilNoExceptions(task, listOfExceptions)` can be implemented.